lib/fdtable.c

```diff
@@ -198,10 +198,12 @@ static ssize_t stdinout_read_vmeth(void *obj, void *buffer, size_t count)
 
 static ssize_t stdinout_write_vmeth(void *obj, const void *buffer, size_t count)
 {
-#ifdef CONFIG_BOARD_NATIVE_POSIX
+#if defined(CONFIG_BOARD_NATIVE_POSIX)
 	return write(1, buffer, count);
```

Comment:
I wonder why this is needed, shouldn't we have a system which "works" as expected regardless of what board we are using?

Reply:
BOARD_NATIVE_POSIX is very, very special. Just consider what happens here: we *define* part of POSIX implementation, but native_posix *runs* on POSIX implementation underneath. And both layers leaks into one another, it's literally a minefield where step aside leads to kaboom. I did a few fixed in that regard, and there're many more to do. So, just things will always be there, just over time, we can find a way to brush it up.